scripts/west_commands/runners/up_squared.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) 2017 Linaro Limited.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+'''Runner for flashing on up squared board via pxe boot.'''
```

Comment:
This really needs to be abstracted and made generic, there is nothing related to the up_squared board here and there is also nothing related to PXE. You are basically just calling a script that can do lots of things, so this runner need to be made more generic. Do not make it appear as it only needed for this specific board or PXE. Basically It can be called ScriptedRunner with an option pointing to a script without having to mention pxe or the up_squared board.

Reply:
I agree that this trend of adding runners which only work for one board (like intel_s1000) should not continue.

If this can't be turned into a generic runner that makes using tftp easier (which would be my preference), I agree that having a generic runner -- name something like CommandRunner because you might not be calling a 'script' -- which requires a binary and some arguments to run is a good way forward.

This would make it easier for special-purpose out of tree boards in addition to up_squared.

I'm thinking something like this for do_add_parser (untested):

```
parser.add_argument('command', help='command to run')
parser.add_argument('extra_args', nargs='*', help='additional command line arguments to pass to command')
```

Then just do `subprocess.check_call(list(args.command) + extra_args)` from do_run or so.

Reply:
I named it as GenericCommandRunner, and it will invoke the command set by the 'command' option.
the needed 'zephyr.strip' is just in the build dir after the west build operation, it is aleready in the parsed cfg argument,  there is no need for another arguments.